src/correlation_service.ts

```diff
@@ -109,44 +114,138 @@ export class CorrelationService implements ICorrelationService {
 
-  public async getByProcessInstanceId(identity: IIdentity, processInstanceId: string): Promise<Correlation> {
+  public async getByProcessInstanceId(identity: IIdentity, processInstanceId: string): Promise<ProcessInstance> {
     await this.ensureUserHasClaim(identity, canReadProcessModelClaim);
 
-    const correlationFromRepo = await this.correlationRepository.getByProcessInstanceId(processInstanceId);
+    const processInstanceFromRepo = await this.correlationRepository.getByProcessInstanceId(processInstanceId);
 
-    const correlation = await this.mapCorrelation(correlationFromRepo.id, [correlationFromRepo]);
+    if (identity.userId !== processInstanceFromRepo.identity.userId) {
+      const userIsSuperAdmin = await this.checkIfUserIsSuperAdmin(identity);
 
-    return correlation;
+      if (!userIsSuperAdmin) {
+        throw new ForbiddenError('Access denied');
+      }
+    }
+
+    const processInstance = await this.mapProcessInstance(processInstanceFromRepo);
+
+    return processInstance;
   }
 
-  public async getSubprocessesForProcessInstance(identity: IIdentity, processInstanceId: string): Promise<Correlation> {
+  public async getSubprocessesForProcessInstance(identity: IIdentity, processInstanceId: string): Promise<Array<ProcessInstance>> {
     await this.ensureUserHasClaim(identity, canReadProcessModelClaim);
 
-    const correlationsFromRepo = await this.correlationRepository.getSubprocessesForProcessInstance(processInstanceId);
+    const processInstancesFromRepo = await this.correlationRepository.getSubprocessesForProcessInstance(processInstanceId);
 
-    const filteredCorrelationsFromRepo = await this.filterCorrelationsFromRepoByIdentity(identity, correlationsFromRepo);
+    const filteredProcessInstancesFromRepo = await this.filterProcessInstancesFromRepoByIdentity(identity, processInstancesFromRepo);
 
-    const noFilteredCorrelations = filteredCorrelationsFromRepo.length === 0;
-    if (noFilteredCorrelations) {
+    if (filteredProcessInstancesFromRepo.length === 0) {
       return undefined;
     }
 
-    const correlation = await this.mapCorrelation(correlationsFromRepo[0].id, correlationsFromRepo);
+    const processInstances =
+      await Promise.map<ProcessInstanceFromRepository, ProcessInstance>(filteredProcessInstancesFromRepo, this.mapProcessInstance.bind(this));
 
-    return correlation;
+    return processInstances;
+  }
+
+  public async getProcessInstancesForCorrelation(
+    identity: IIdentity,
+    correlationId: string,
+    offset?: number,
+    limit?: number,
+  ): Promise<Array<ProcessInstance>> {
+    await this.ensureUserHasClaim(identity, canReadProcessModelClaim);
+
+    const processInstancesFromRepo = await this.correlationRepository.getByCorrelationId(correlationId);
+
+    const noCorrelationsFound = !processInstancesFromRepo || processInstancesFromRepo.length === 0;
+    if (noCorrelationsFound) {
+      throw new NotFoundError(`Correlation with id "${correlationId}" not found.`);
```

Comment:
Die anderen beiden Methoden werfen einen Fehler der mit `No ProcessInstances for ...`. Ich würde hier auch auf die ProcessInstances eingehen statt auf die Correlation mit der gesucht wird.

```suggestion
      throw new NotFoundError(`No ProcessInstances for Correlation with id "${correlationId}" found.`);
```